src/components/input.js

```diff
@@ -146,11 +146,14 @@ export const SearchInput = ({ value, onChange, ...props }) => {
 export const DelayedSearchInput = withDebouncedChange(SearchInput)
 
 export const NumberInput = ({ onChange, onBlur, min = -Infinity, max = Infinity, onlyInteger = false, isClearable = true, tooltip, value, ...props }) => {
-  Utils.useConsoleAssert(props.id || props['aria-label'], 'In order to be accessible, NumberInput needs a label')
+  // If the user provided a tooltip but no other label, use the tooltip as the label for the input
+  Utils.useLabelAssert('NumberInput', { tooltip, ...props, allowTooltip: true })
+
   const [internalValue, setInternalValue] = useState()
 
   const numberInputChild = div([input(_.merge({
     type: 'number',
+    'aria-label': Utils.useAriaLabelOrTooltip({ tooltip, ...props }),
```

Comment:
move `aria-label` to the top of the props list

Reply:
Like the others, `type` is pretty tightly bound to the form element and dictates how it will appear and behavior from an accessibility perspective. But again, happy to bump the `aria-*` properties above it if you insist.